client/client.go

```diff
@@ -923,16 +926,28 @@ func (r *NotaryRepository) RotateKey(role string, serverManagesKey bool) error {
 	}
 
 	var (
-		pubKey    data.PublicKey
-		err       error
-		errFmtMsg string
+		pubKey     data.PublicKey
+		pubKeyList data.KeyList
+		err        error
+		errFmtMsg  string
 	)
-	switch serverManagesKey {
-	case true:
+
+	switch {
```

Comment:
There are three distinct cases: the server manages the key, the server doesn't manage the key and keys to rotate to have been provided, or the server doesn't manage the key and no keys were provided.

I don't follow how if/else cleans this up. Is it just a style preference?